python/ray/train/tests/test_gpu.py

```diff
@@ -307,6 +307,32 @@ def assert_env_var_set():
     worker_group.execute(assert_env_var_set)
 
 
+def test_torch_fail_on_nccl_timeout(ray_start_4_cpus_2_gpus):
+    """Tests that TorchTrainer raises exception on NCCL timeouts."""
+
+    def train_fn():
+        model = torch.nn.Linear(1, 1)
+        model = train.torch.prepare_model(model)
+
+        # Rank 0 worker will never reach the collective operation.
+        # NCCL should timeout.
+        if session.get_world_rank() == 0:
+            while True:
+                pass
```

Comment:
nit: can we have a `time.sleep` here? that way we don't consume an entire CPU unnecessarily 🔥